app/Service/User/UserService.php

```diff
@@ -1,135 +1,88 @@
 
 class UserService
 {
-
-	private string $key_hash_mac = "";
-	private int $expirationTime_s = 60*30; //30 minutes expiration time
+	private int $expirationTime_s = 60 * 30; //30 minutes expiration time
 	private string $JWTokenRedisPre = 'user:0:JWToken_';
-	
+
 	private $redis;
-	
+
 	public function __construct()
 	{
-		$this->key_hash_mac = env('KEY_HASH_MAC', "CHANGE_MY_IN_ENV_Password");		
 		$this->redis = new Redis();
-		
 	}
 
-	public function checkUserCredentials(string $userDNI, string $password): bool
+	public function checkUserCredentials(User $user, string $password): bool
 	{
-		$user = User::where('dni', $userDNI)->first();
-		
-		if(empty($user) == False && empty($user->password) == False){
-			try{
-				if(password_verify($password, $user->password)){
-					return True;
-				}else{
-					return False;
-				}
-				
-			}catch (Exception $e){
-				return False;
-			}
-		}else{
-			return False;
+		if (!$user || !Hash::check($password, $user->password)) {
+			throw new InvalidCredentialsException();
 		}
+		return true;
 	}
-	
-	public function getUserIDByDNI(string $userDNI): string  | int | bool
+
+	public function getUserByDNI(string $userDNI): User
 	{
 		$user = User::where('dni', $userDNI)->first();
-		
-		if(empty($user) == False && empty($user->id) == False){
-			return $user->id;
-		}else{
-			return False;
+
+		if (!$user) {
+			throw new UserNotFoundException($userDNI);
 		}
+		return $user;
 	}
-	
-	
-	//public function generateJWToken(string $userID, string $issuerID)
-	public function generateJWToken(string | int $userID): string
+
+	public function generateJWToken(User $user): string
 	{
-		if(is_numeric($userID)){
-			$userID = (string)$userID;
-		}
-		
-		
-		// Create token header as a JSON string
-		$header = json_encode(['typ' => 'JWT', 'alg' => 'HS256']);
-		
-		
-		// Create token payload as a JSON string
-		$payload = json_encode(['user_id' => $userID]);
-		
-		// Encode Header to Base64Url String
-		$base64UrlHeader = str_replace(['+', '/', '='], ['-', '_', ''], base64_encode($header));
-
-		// Encode Payload to Base64Url String
-		$base64UrlPayload = str_replace(['+', '/', '='], ['-', '_', ''], base64_encode($payload));
-
-		// Create Signature Hash
-		//https://www.php.net/manual/en/function.hash-hmac.php
-		$signature = hash_hmac('sha256', $base64UrlHeader . "." . $base64UrlPayload, $this->key_hash_mac, true);
-		
-		// Encode Signature to Base64Url String
-		$base64UrlSignature = str_replace(['+', '/', '='], ['-', '_', ''], base64_encode($signature));
-
-		// Create JWT
-		$jwt = $base64UrlHeader . "." . $base64UrlPayload . "." . $base64UrlSignature;
+		$jwt = $user->createToken('loginToken')->accessToken;
 
+		if (!$jwt) {
+			throw new CouldNotCreateJWTokenPassportException($user->id);
+		}
 		return $jwt;
 	}
 
-	
-
-	public function storeUserIDAndTokenRedis( string | int $userID, string $token): bool
+	public function storeUserIDAndTokenRedis(string | int $userID, string $token): bool
```

Comment:
string $userID

Reply:
Done and removed the if statement to check if ID is_numeric, no longer needed.